include/gatekeeper_lls.h

```diff
@@ -191,6 +196,29 @@ struct lls_config {
 	uint16_t           tx_queue_back;
 };
 
+/*
+ * Functions for resolving IPv4 addresses to link layer addresses.
+ *
+ * resolve_arp() calls the given callback function once when a
+ * resolution is available.
+ *
+ * hold_arp() calls the given callback function function when a
+ * resolution is available and whenever there is an update.
+ *
+ * put_arp() revokes the subscription. If a block calls either
+ * resolve_arp() or hold_arp(), it should later call put_arp()
+ * when it is no longer interested in the map. Callers of
+ * resolve_arp() may choose to put_arp() on a map after some
+ * timeout if no resolution via the callack function is received.
+ *
+ * For all the ARP functions, @ip_be must be in network format.
+ */
+int resolve_arp(lls_update_cb cb, void *arg, struct in_addr *ip_be,
+	unsigned int lcore_id);
+int hold_arp(lls_update_cb cb, void *arg, struct in_addr *ip_be,
```

Comment:
To check my understanding about the ``hold_arp`` function, to allow the GK to obtain the entries, we need to cache the entries in the GK block in the callback function, is this right?